dbms/Common/HashTable/StringHashTable.h

```diff
@@ -237,6 +241,13 @@ class StringHashTable : private boost::noncopyable
             return func(self.m0, VoidKey{}, 0);
         }
 
+        if (x.data[sz - 1] == 0)
```

Comment:
hmm, this is very critical path for performance. Let's see how perftest goes.

Reply:
The branch is easy predictable and there is a hope that the memory load and arithmetic will be executed with further operations in instruction parallel manner... But I'm not sure :)